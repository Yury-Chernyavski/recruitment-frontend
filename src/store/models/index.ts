export { type ITodo } from "./ITodo";
export { type ITodoState } from "./ITodoState";
