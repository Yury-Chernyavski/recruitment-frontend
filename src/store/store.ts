import { configureStore } from "@reduxjs/toolkit";
import todoSlice from "./reducers/todoSlice";
import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux";


export const store = configureStore({
  reducer: {
    todos: todoSlice
  }
});


export type RootSate = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
export const useAppDispatch: () => AppDispatch = useDispatch;
export const useAppSelector: TypedUseSelectorHook<RootSate> = useSelector;
