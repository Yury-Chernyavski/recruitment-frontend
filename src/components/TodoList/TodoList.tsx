import React from "react";
import { getTodoList } from "../../store/selectors/getTodoList";
import { useAppSelector } from "../../store/store";
import { TodoItem } from "../../components";
import styles from "./TodoList.module.scss";

export const TodoList = () => {
  const todoList = useAppSelector(getTodoList);

  return (
    <ul
      className={styles.todosList}
      data-testid="todoList"
    >
      {todoList.map(t => (
        !t.isChecked && <TodoItem
          key={t.id}
          id={t.id}
          isChecked={t.isChecked}
          text={t.text}
        />
      ))}
      {todoList.map(t => (
        t.isChecked && <TodoItem
          key={t.id}
          id={t.id}
          isChecked={t.isChecked}
          text={t.text}
        />
      ))}
    </ul>
  );
};
