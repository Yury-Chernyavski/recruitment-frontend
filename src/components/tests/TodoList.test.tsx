import React from "react";
import * as Redux from "react-redux";
// import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import { TodoList } from "../TodoList/TodoList";
import { store } from "../../store/store";

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));

const todos = [
  { id: 1, isChecked: false, text: "Task 1" },
  { id: 2, isChecked: true, text: "Task 2" },
];

describe("TodoList Component", () => {
  it("matches snapshot before useSelector is called", () => {
    const useSelectorSpy = jest.spyOn(Redux, "useSelector");
    useSelectorSpy.mockReturnValue([]);
    const { asFragment } = render(
      <Redux.Provider store={store}>
        <TodoList />
      </Redux.Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot after useSelector is called", () => {
    const useSelectorSpy = jest.spyOn(Redux, "useSelector");
    useSelectorSpy.mockReturnValue(todos);
    const { asFragment } = render(
      <Redux.Provider store={store}>
        <TodoList />
      </Redux.Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
