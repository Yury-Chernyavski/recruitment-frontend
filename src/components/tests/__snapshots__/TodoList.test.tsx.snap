// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList Component matches snapshot after useSelector is called 1`] = `
<DocumentFragment>
  <ul
    class="todosList"
    data-testid="todoList"
  >
    <li
      class="item"
    >
      <input
        class="check"
        type="checkbox"
      />
      <span>
        Task 1
      </span>
      <button
        class="deleteButton"
      >
        Delete
      </button>
    </li>
    <li
      class="item"
    >
      <input
        checked=""
        class="check"
        type="checkbox"
      />
      <span>
        Task 2
      </span>
      <button
        class="deleteButton"
      >
        Delete
      </button>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`TodoList Component matches snapshot before useSelector is called 1`] = `
<DocumentFragment>
  <ul
    class="todosList"
    data-testid="todoList"
  />
</DocumentFragment>
`;
